sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReliableDownloadFlux.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation;
+
+import reactor.core.CoreSubscriber;
+import reactor.core.publisher.Flux;
+
+import java.nio.ByteBuffer;
+import java.util.function.BiFunction;
+import java.util.function.Supplier;
+
+/**
+ * A {@code Flux<ByteBuffer>} implementation which is capable of performing a reliable download by applying a resume
+ * operation if an error occurs during the download.
+ */
+public final class ReliableDownloadFlux extends Flux<ByteBuffer> {
+    private final Supplier<Flux<ByteBuffer>> downloadSupplier;
+    private final BiFunction<Throwable, Long, Flux<ByteBuffer>> onDownloadErrorResume;
+    private final int maxRetries;
+
+    private long position;
```

Comment:
Since this type is stateful, what happens when there are multiple subscribers to this flux? The position can be unsafely modified by failures observed by different subscribers.

Reply:
Made it so the subscription will being with the class level position and then maintain it using a closure capture. Same with the retryCount.